### ISSUE TITLE
Tux's physics are extremely broken

### ISSUE THREAD

**Issue by weluvgoatz** [SuperTux/supertux#1120]
I am on the WIP 0.6.1, but some of these errors appeared in the 0.6.0 release.

**0.6 Bug:**

When Tux falls, he falls very slowly and the game is slowed down until he stops.

**0.6.1 Bugs:**

Tux can hardly jump, he jumps 2-3 tiles instead of 4-5 and abruptly falls. When he jumps and falls, the game lags heavily and he jitters when he falls.

Because of this I cannot complete any level because Tux cannot jump over pits.

**Comment by Rusty-Box** [comment]
The game indeed is lagging at the moment. However I cannot reproduce the bug involving Tux's jump height.

**Comment by weluvgoatz** [comment]
The jump height was an issue with my computer, sorry.  But jumping, falling, and turning on worldmaps lags the game significantly. Also, if I haven't set my graphics settings to extremely low, the menus open extremely slowly.  Also, I want the game to be able to save my graphics settings (like setting fps to 30)

**Comment by Alzter** [comment]
Ever since 0.6.0 I've been noticing the game takes up a lot more CPU power and sometimes lags more.

**Comment by HybridDog** [comment]
The Framerate setting is currently broken; supertux doesn't work right if it's not set to 60.
I think that I changed screen_manager.cpp in an old supertux version first, and later, while creating that PR, I messed up the framerate setting.
https://github.com/SuperTux/supertux/pull/1055/files
The update_gamelogic function takes a dtime as argument, so I assumed that it supports variable dtime values. In some places, e.g. worldmap camera panning, the dtime is simply ignored (#1100 should fix it).
When I press F2 and significantly increase the game speed, Tux can no longer jump high, i.e. jumping probably does not work right with a large dtime.
Before the PR (e.g. in version 6.0), dtime was 1/framerate, so at that time, the framerate option probably had an impact on jump height, too.

When I made that PR, I didn't think about triple buffering yet. I presume that a high elapsed time can be followed by a low elapsed time, then the frames generated for those dtimes are stored in the buffer and after that vsync alignes both frames with 1/60 time difference.
Here are changes to compensate for this aligning when vsync is enabled:
https://github.com/HybridDog/supertux/commit/c815527316f1d3926c7a7225b167cc964ed39382
This could fix #1117.

To fix the jump height, maybe update_gamelogic can be changed to do more than one collision calculation when dtime is high.
